src/main/java/com/openshift/jenkins/plugins/pipeline/model/IOpenShiftImageTagger.java

```diff
@@ -132,113 +122,170 @@ default String deriveImageTag(String imageID, IImageStream srcIS) {
-			client.update(destIS);
-			
-			
+
+			List<String> newDestinationStreams = Arrays.asList( destStreams.split( "," ) );
+			List<String> newDestTags = Arrays.asList( destTags.split( "," ) );
+
+			/**
+			 * The following logic is permissive of the following combinations
+			 *  (1) 1 dest stream and 1 dest tag  	(i.e. a single new stream:tag combination)
+			 * 	(2) 1 dest stream and N dest tags   (i.e. all tags apply to the same stream)
+			 * 	(3) 1 dest tag and N dest streams	(i.e. all streams will get the same tag)
+			 * 	(4) N dest streams and N dest tags	(i.e. each index will be a unique stream:tag)
+			 */
+
+			if ( newDestinationStreams.size() != 1 && newDestTags.size() != 1 && newDestinationStreams.size() != newDestTags.size() ) {
+				String error = String.format( "Destination streams (%s) cardinality [%d] is incompatible with destination tag (%s) cardinality [%d]", newDestinationStreams, newDestinationStreams.size(), newDestTags, newDestTags.size() );
+				throw new IllegalArgumentException( error );
+			}
+
+			Iterator<String> iDestStreams = newDestinationStreams.iterator();
+			Iterator<String> iDestTags = newDestTags.iterator();
+			String destStream = null;
+			String destTag = null;
+			while ( iDestStreams.hasNext() || iDestTags.hasNext() ) {
+
+				if ( iDestStreams.hasNext() ) {
+					destStream = iDestStreams.next().trim();
+				}
+
+				if ( iDestTags.hasNext() ) {
+					destTag = iDestTags.next().trim();
+				}
+
+				IImageStream destIS = null;
+				for ( int retries = 10; retries > 0; retries-- ) {
+					try {
+						destIS = destClient.get(ResourceKind.IMAGE_STREAM, destStream, destNS);
+					} catch (com.openshift.restclient.OpenShiftException e) {
+						String createJson = "{\"kind\": \"ImageStream\",\"apiVersion\": \"v1\",\"metadata\": {\"name\": \"" +
+								destStream + "\",\"creationTimestamp\": null},\"spec\": {},\"status\": {\"dockerImageRepository\": \"\"}}";
+
+						Map<String,String> newOverrides = new HashMap<String,String>(overrides);
+						// we don't want this step's source namespace to be the creator's namespace,
+						// but rather this step's destination namespace, so clear out the override
+						// and then reuse all other overrides.
+						newOverrides.remove("namespace");
+
+						String token = destToken;
+						if ( token == null || token.isEmpty() ) {
+							token = getAuthToken();
+						}
+
+						OpenShiftCreator isCreator = new OpenShiftCreator(getApiURL(newOverrides), destNS, token, ""+chatty, createJson);
+						isCreator.setAuth(Auth.createInstance(chatty ? listener : null, getApiURL(newOverrides), overrides));
+
+						boolean newISCreated = isCreator.coreLogic(launcher, listener, newOverrides);
+
+						if (!newISCreated) {
+							listener.getLogger().println(String.format(MessageConstants.EXIT_TAG_CANNOT_CREATE_DEST_IS, destStream, destNS));
+							return false;
+						}
+
+						for ( int retries2 = 10; retries2 > 0; retries2-- ) {
```

Comment:
Yeah a get should be OK vs. the update ... let's remove / have a little less clutter.